tests/test_target.py

```diff
@@ -62,15 +95,26 @@ def test_read_multiline_values(target):
         'test3': '3\n\n4\n\n',
     }
 
-    tempdir = tempfile.mkdtemp(prefix='devlib-test-')
-    for key, value in data.items():
-        path = os.path.join(tempdir, key)
-        with open(path, 'w', encoding='utf-8') as wfh:
-            wfh.write(value)
+    print(f'target={target.__class__.__name__} os={target.os} hostname={target.hostname}')
+
+    with target.make_temp() as tempdir:
+        print(f'Created {tempdir}.')
+
+        for key, value in data.items():
+            path = os.path.join(tempdir, key)
+            print(f'Writing {value!r} to {path}...')
+            target.write_value(path, value, verify=False,
+                               as_root=target.conn.connected_as_root)
+
+        print('Reading values from target...')
+        raw_result = target.read_tree_values_flat(tempdir)
+        result = {os.path.basename(k): v for k, v in raw_result.items()}
 
-    raw_result = target.read_tree_values_flat(tempdir)
-    result = {os.path.basename(k): v for k, v in raw_result.items()}
+    print(f'Removing {target.working_directory}...')
+    target.remove(target.working_directory)
 
-    shutil.rmtree(tempdir)
+    if target_runner is not None:
```

Comment:
the `None` test can be dropped once the case disappears.

Reply:
Done.